digital_image_processing/dithering/burkes.py

```diff
@@ -0,0 +1,87 @@
+"""
+Implementation Burke's algorithm (dithering)
+"""
+from cv2 import destroyAllWindows, imread, imshow, waitKey
+import numpy as np
+
+
+class Burkes:
+    """
+    Burke's algorithm is using for converting grayscale image to black and white version
+    Source: Source: https://en.wikipedia.org/wiki/Dither
+
+    Note:
+        * Best results are given with threshold= ~1/2 * max greyscale value.
+        * This implementation get RGB image and converts it to greyscale in runtime.
+    """
+
+    def __init__(self, input_img, threshold: int):
+        self.min_threshold = 0
+        # max greyscale value for #FFFFFF
+        self.max_threshold = int(self.get_greyscale(255, 255, 255))
+
+        if not self.min_threshold < threshold < self.max_threshold:
+            raise ValueError(f"Factor value should be from 0 to {self.max_threshold}")
+
+        self.input_img = input_img
+        self.threshold = threshold
+        self.width, self.height = self.input_img.shape[1], self.input_img.shape[0]
+
+        # error table size (+4 columns and +1 row) greater than input image because of
+        # lack of if statements
+        self.error_table = [
+            [0 for _ in range(self.height + 4)] for __ in range(self.width + 1)
+        ]
+        self.output_img = np.ones((self.width, self.height, 3), np.uint8) * 255
+
+    @classmethod
+    def get_greyscale(cls, blue: int, green: int, red: int) -> float:
+        """
+        >>> Burkes.get_greyscale(5, 4, 3)
+        3.753
+        """
+        return 0.2126 * red + 0.587 * green + 0.114 * blue
```

Comment:
If line 38 is in blue, green, red order then line 43 should also be in blue, green, red order.

In most instances, these numbers are stored in red, green, blue order.